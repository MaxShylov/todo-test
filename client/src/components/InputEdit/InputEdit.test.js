import React from 'react';
import { shallow } from 'enzyme'

import { InputEdit } from './InputEdit.jsx'


describe('InputEdit', () => {

  const props = {
    id: 1,
    content: 'qwe'
  };

  describe('initial', () => {
    const wrapper = shallow(<InputEdit {...props} />);

    it('renders properly', () => expect(wrapper).toMatchSnapshot());
  });

  describe('method onChange', () => {
    const
      wrapper = shallow(<InputEdit {...props} />),
      value = '123';

    wrapper.setState({
      isEdit: true,
      value: ''
    });

    wrapper.find('TextArea').simulate('change', { target: { value } });

    it('renders properly', () => expect(wrapper).toMatchSnapshot());
    it('state.value updated', () => expect(wrapper.state().value).toEqual(value));
  });

  describe('method updateTask without state.value', () => {
    const
      mockUpdateTask = jest.fn(),
      mockSaveTasks = jest.fn(),
      nextProps = {
        ...props,
        updateTask: mockUpdateTask,
        saveTasks: mockSaveTasks
      },
      wrapper = shallow(<InputEdit {...nextProps} />);

    wrapper.setState({
      isEdit: true,
      value: ''
    });

    wrapper.instance().updateTask();

    it('renders properly', () => expect(wrapper).toMatchSnapshot());
    it('state.value updated to props.content', () => expect(wrapper.state().value).toEqual(nextProps.content));
    it('updateTask didn\'t call', () => expect(mockUpdateTask).not.toHaveBeenCalled());
    it('saveTasks didn\'t call', () => expect(mockSaveTasks).not.toHaveBeenCalled());
  });

  describe('method updateTask with state.value', () => {
    const
      mockUpdateTask = jest.fn(),
      mockSaveTasks = jest.fn(),
      nextProps = {
        ...props,
        updateTask: mockUpdateTask,
        saveTasks: mockSaveTasks
      },
      wrapper = shallow(<InputEdit {...nextProps} />);

    wrapper.setState({
      isEdit: true,
      value: 'qwe'
    });

    wrapper.instance().updateTask();

    it('renders properly', () => expect(wrapper).toMatchSnapshot());
    it('state.isEdit updated', () => expect(wrapper.state().isEdit).toEqual(false));
    it('updateTask called', () => expect(mockUpdateTask).toHaveBeenCalled());
    it('saveTasks called', () => expect(mockSaveTasks).toHaveBeenCalled());
  });

  describe('method openEdit', () => {
    const wrapper = shallow(<InputEdit {...props} />);

    wrapper.setState({ isEdit: false });

    wrapper.find('Button').simulate('click');

    it('renders properly', () => expect(wrapper).toMatchSnapshot());
    it('state.isEdit updated', () => expect(wrapper.state().isEdit).toEqual(true));
  });

  describe('method closeEdit', () => {
    const wrapper = shallow(<InputEdit {...props} />);

    wrapper.setState({
      isEdit: true,
      value: 'asd'
    });

    wrapper.find('Button').simulate('click');

    it('renders properly', () => expect(wrapper).toMatchSnapshot());
    it('state.isEdit updated', () => expect(wrapper.state().isEdit).toEqual(false));
    it('state.value updated', () => expect(wrapper.state().value).toEqual(props.content));
  });

});
