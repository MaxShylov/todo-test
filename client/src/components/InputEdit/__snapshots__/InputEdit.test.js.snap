// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputEdit initial renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputEdit
    content="qwe"
    id={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="text"
        >
          qwe
        </div>,
        <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Open edit"
          transitionName="zoom-big-fast"
        >
          <Button
            block={false}
            ghost={false}
            icon="edit"
            loading={false}
            onClick={[Function]}
            type="primary"
          />
        </Tooltip>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "qwe",
          "className": "text",
        },
        "ref": null,
        "rendered": "qwe",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrowPointAtCenter": false,
          "autoAdjustOverflow": true,
          "children": <Button
            block={false}
            ghost={false}
            icon="edit"
            loading={false}
            onClick={[Function]}
            type="primary"
          />,
          "mouseEnterDelay": 0.1,
          "mouseLeaveDelay": 0.1,
          "placement": "top",
          "title": "Open edit",
          "transitionName": "zoom-big-fast",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "ghost": false,
            "icon": "edit",
            "loading": false,
            "onClick": [Function],
            "type": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="text"
          >
            qwe
          </div>,
          <Tooltip
            arrowPointAtCenter={false}
            autoAdjustOverflow={true}
            mouseEnterDelay={0.1}
            mouseLeaveDelay={0.1}
            placement="top"
            title="Open edit"
            transitionName="zoom-big-fast"
          >
            <Button
              block={false}
              ghost={false}
              icon="edit"
              loading={false}
              onClick={[Function]}
              type="primary"
            />
          </Tooltip>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "qwe",
            "className": "text",
          },
          "ref": null,
          "rendered": "qwe",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "arrowPointAtCenter": false,
            "autoAdjustOverflow": true,
            "children": <Button
              block={false}
              ghost={false}
              icon="edit"
              loading={false}
              onClick={[Function]}
              type="primary"
            />,
            "mouseEnterDelay": 0.1,
            "mouseLeaveDelay": 0.1,
            "placement": "top",
            "title": "Open edit",
            "transitionName": "zoom-big-fast",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "ghost": false,
              "icon": "edit",
              "loading": false,
              "onClick": [Function],
              "type": "primary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`InputEdit method closeEdit renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputEdit
    content="qwe"
    id={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="text"
        >
          qwe
        </div>,
        <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Open edit"
          transitionName="zoom-big-fast"
        >
          <Button
            block={false}
            ghost={false}
            icon="edit"
            loading={false}
            onClick={[Function]}
            type="primary"
          />
        </Tooltip>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "qwe",
          "className": "text",
        },
        "ref": null,
        "rendered": "qwe",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrowPointAtCenter": false,
          "autoAdjustOverflow": true,
          "children": <Button
            block={false}
            ghost={false}
            icon="edit"
            loading={false}
            onClick={[Function]}
            type="primary"
          />,
          "mouseEnterDelay": 0.1,
          "mouseLeaveDelay": 0.1,
          "placement": "top",
          "title": "Open edit",
          "transitionName": "zoom-big-fast",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "ghost": false,
            "icon": "edit",
            "loading": false,
            "onClick": [Function],
            "type": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="text"
          >
            qwe
          </div>,
          <Tooltip
            arrowPointAtCenter={false}
            autoAdjustOverflow={true}
            mouseEnterDelay={0.1}
            mouseLeaveDelay={0.1}
            placement="top"
            title="Open edit"
            transitionName="zoom-big-fast"
          >
            <Button
              block={false}
              ghost={false}
              icon="edit"
              loading={false}
              onClick={[Function]}
              type="primary"
            />
          </Tooltip>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "qwe",
            "className": "text",
          },
          "ref": null,
          "rendered": "qwe",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "arrowPointAtCenter": false,
            "autoAdjustOverflow": true,
            "children": <Button
              block={false}
              ghost={false}
              icon="edit"
              loading={false}
              onClick={[Function]}
              type="primary"
            />,
            "mouseEnterDelay": 0.1,
            "mouseLeaveDelay": 0.1,
            "placement": "top",
            "title": "Open edit",
            "transitionName": "zoom-big-fast",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "ghost": false,
              "icon": "edit",
              "loading": false,
              "onClick": [Function],
              "type": "primary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`InputEdit method onChange renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputEdit
    content="qwe"
    id={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <TextArea
          autosize={true}
          disabled={false}
          onChange={[Function]}
          onPressEnter={[Function]}
          value="123"
        />,
        <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Close edit"
          transitionName="zoom-big-fast"
        >
          <Button
            block={false}
            ghost={false}
            icon="close"
            loading={false}
            onClick={[Function]}
            type="primary"
          />
        </Tooltip>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autosize": true,
          "disabled": false,
          "onChange": [Function],
          "onPressEnter": [Function],
          "value": "123",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrowPointAtCenter": false,
          "autoAdjustOverflow": true,
          "children": <Button
            block={false}
            ghost={false}
            icon="close"
            loading={false}
            onClick={[Function]}
            type="primary"
          />,
          "mouseEnterDelay": 0.1,
          "mouseLeaveDelay": 0.1,
          "placement": "top",
          "title": "Close edit",
          "transitionName": "zoom-big-fast",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "ghost": false,
            "icon": "close",
            "loading": false,
            "onClick": [Function],
            "type": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <TextArea
            autosize={true}
            disabled={false}
            onChange={[Function]}
            onPressEnter={[Function]}
            value="123"
          />,
          <Tooltip
            arrowPointAtCenter={false}
            autoAdjustOverflow={true}
            mouseEnterDelay={0.1}
            mouseLeaveDelay={0.1}
            placement="top"
            title="Close edit"
            transitionName="zoom-big-fast"
          >
            <Button
              block={false}
              ghost={false}
              icon="close"
              loading={false}
              onClick={[Function]}
              type="primary"
            />
          </Tooltip>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autosize": true,
            "disabled": false,
            "onChange": [Function],
            "onPressEnter": [Function],
            "value": "123",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "arrowPointAtCenter": false,
            "autoAdjustOverflow": true,
            "children": <Button
              block={false}
              ghost={false}
              icon="close"
              loading={false}
              onClick={[Function]}
              type="primary"
            />,
            "mouseEnterDelay": 0.1,
            "mouseLeaveDelay": 0.1,
            "placement": "top",
            "title": "Close edit",
            "transitionName": "zoom-big-fast",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "ghost": false,
              "icon": "close",
              "loading": false,
              "onClick": [Function],
              "type": "primary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`InputEdit method openEdit renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputEdit
    content="qwe"
    id={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <TextArea
          autosize={true}
          disabled={false}
          onChange={[Function]}
          onPressEnter={[Function]}
          value="qwe"
        />,
        <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Close edit"
          transitionName="zoom-big-fast"
        >
          <Button
            block={false}
            ghost={false}
            icon="close"
            loading={false}
            onClick={[Function]}
            type="primary"
          />
        </Tooltip>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autosize": true,
          "disabled": false,
          "onChange": [Function],
          "onPressEnter": [Function],
          "value": "qwe",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrowPointAtCenter": false,
          "autoAdjustOverflow": true,
          "children": <Button
            block={false}
            ghost={false}
            icon="close"
            loading={false}
            onClick={[Function]}
            type="primary"
          />,
          "mouseEnterDelay": 0.1,
          "mouseLeaveDelay": 0.1,
          "placement": "top",
          "title": "Close edit",
          "transitionName": "zoom-big-fast",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "ghost": false,
            "icon": "close",
            "loading": false,
            "onClick": [Function],
            "type": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <TextArea
            autosize={true}
            disabled={false}
            onChange={[Function]}
            onPressEnter={[Function]}
            value="qwe"
          />,
          <Tooltip
            arrowPointAtCenter={false}
            autoAdjustOverflow={true}
            mouseEnterDelay={0.1}
            mouseLeaveDelay={0.1}
            placement="top"
            title="Close edit"
            transitionName="zoom-big-fast"
          >
            <Button
              block={false}
              ghost={false}
              icon="close"
              loading={false}
              onClick={[Function]}
              type="primary"
            />
          </Tooltip>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autosize": true,
            "disabled": false,
            "onChange": [Function],
            "onPressEnter": [Function],
            "value": "qwe",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "arrowPointAtCenter": false,
            "autoAdjustOverflow": true,
            "children": <Button
              block={false}
              ghost={false}
              icon="close"
              loading={false}
              onClick={[Function]}
              type="primary"
            />,
            "mouseEnterDelay": 0.1,
            "mouseLeaveDelay": 0.1,
            "placement": "top",
            "title": "Close edit",
            "transitionName": "zoom-big-fast",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "ghost": false,
              "icon": "close",
              "loading": false,
              "onClick": [Function],
              "type": "primary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`InputEdit method updateTask with state.value renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputEdit
    content="qwe"
    id={1}
    saveTasks={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    updateTask={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "content": "qwe",
              "id": 1,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="text"
        >
          qwe
        </div>,
        <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Open edit"
          transitionName="zoom-big-fast"
        >
          <Button
            block={false}
            ghost={false}
            icon="edit"
            loading={false}
            onClick={[Function]}
            type="primary"
          />
        </Tooltip>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "qwe",
          "className": "text",
        },
        "ref": null,
        "rendered": "qwe",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrowPointAtCenter": false,
          "autoAdjustOverflow": true,
          "children": <Button
            block={false}
            ghost={false}
            icon="edit"
            loading={false}
            onClick={[Function]}
            type="primary"
          />,
          "mouseEnterDelay": 0.1,
          "mouseLeaveDelay": 0.1,
          "placement": "top",
          "title": "Open edit",
          "transitionName": "zoom-big-fast",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "ghost": false,
            "icon": "edit",
            "loading": false,
            "onClick": [Function],
            "type": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="text"
          >
            qwe
          </div>,
          <Tooltip
            arrowPointAtCenter={false}
            autoAdjustOverflow={true}
            mouseEnterDelay={0.1}
            mouseLeaveDelay={0.1}
            placement="top"
            title="Open edit"
            transitionName="zoom-big-fast"
          >
            <Button
              block={false}
              ghost={false}
              icon="edit"
              loading={false}
              onClick={[Function]}
              type="primary"
            />
          </Tooltip>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "qwe",
            "className": "text",
          },
          "ref": null,
          "rendered": "qwe",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "arrowPointAtCenter": false,
            "autoAdjustOverflow": true,
            "children": <Button
              block={false}
              ghost={false}
              icon="edit"
              loading={false}
              onClick={[Function]}
              type="primary"
            />,
            "mouseEnterDelay": 0.1,
            "mouseLeaveDelay": 0.1,
            "placement": "top",
            "title": "Open edit",
            "transitionName": "zoom-big-fast",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "ghost": false,
              "icon": "edit",
              "loading": false,
              "onClick": [Function],
              "type": "primary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`InputEdit method updateTask without state.value renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputEdit
    content="qwe"
    id={1}
    saveTasks={[MockFunction]}
    updateTask={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="text"
        >
          qwe
        </div>,
        <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Open edit"
          transitionName="zoom-big-fast"
        >
          <Button
            block={false}
            ghost={false}
            icon="edit"
            loading={false}
            onClick={[Function]}
            type="primary"
          />
        </Tooltip>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "qwe",
          "className": "text",
        },
        "ref": null,
        "rendered": "qwe",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrowPointAtCenter": false,
          "autoAdjustOverflow": true,
          "children": <Button
            block={false}
            ghost={false}
            icon="edit"
            loading={false}
            onClick={[Function]}
            type="primary"
          />,
          "mouseEnterDelay": 0.1,
          "mouseLeaveDelay": 0.1,
          "placement": "top",
          "title": "Open edit",
          "transitionName": "zoom-big-fast",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "ghost": false,
            "icon": "edit",
            "loading": false,
            "onClick": [Function],
            "type": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="text"
          >
            qwe
          </div>,
          <Tooltip
            arrowPointAtCenter={false}
            autoAdjustOverflow={true}
            mouseEnterDelay={0.1}
            mouseLeaveDelay={0.1}
            placement="top"
            title="Open edit"
            transitionName="zoom-big-fast"
          >
            <Button
              block={false}
              ghost={false}
              icon="edit"
              loading={false}
              onClick={[Function]}
              type="primary"
            />
          </Tooltip>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "qwe",
            "className": "text",
          },
          "ref": null,
          "rendered": "qwe",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "arrowPointAtCenter": false,
            "autoAdjustOverflow": true,
            "children": <Button
              block={false}
              ghost={false}
              icon="edit"
              loading={false}
              onClick={[Function]}
              type="primary"
            />,
            "mouseEnterDelay": 0.1,
            "mouseLeaveDelay": 0.1,
            "placement": "top",
            "title": "Open edit",
            "transitionName": "zoom-big-fast",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "ghost": false,
              "icon": "edit",
              "loading": false,
              "onClick": [Function],
              "type": "primary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
